contentcuration/contentcuration/serializers.py

```diff
@@ -762,11 +762,27 @@ class PublicChannelSerializer(ChannelFieldMixin, serializers.ModelSerializer):
     """
     kind_count = serializers.SerializerMethodField('generate_kind_count')
     matching_tokens = serializers.SerializerMethodField('match_tokens')
+    icon_encoding = serializers.SerializerMethodField('get_thumbnail_encoding')
 
     def match_tokens(self, channel):
         tokens = json.loads(channel.tokens) if hasattr(channel, 'tokens') else []
         return list(channel.secret_tokens.filter(token__in=tokens).values_list('token', flat=True))
 
+    def get_thumbnail_encoding(self, channel):
+        """
+        Historically, we did not set channel.icon_encoding in the Studio database. We
+        only set it in the exported Kolibri sqlite db. So when Kolibri asks for the channel
+        information, fall back to the channel thumbnail data if icon_encoding is not set.
+        """
+        if channel.icon_encoding:
+            return channel.icon_encoding
```

Comment:
Is it always safe to just return the thumbnail_encoding? If so this could be removed.

Reply:
This is so that the thumbnail used when the channel was published remains the same despite the user changing the thumbnail after publishing

Reply:
It currently isn't being set, although from my reading of the code I think that was unintentional and may not have always been the case, so I didn't want to break the old behavior without more discussion about whether or not we want to use this field going forward.